wp-tevko-responsive-images.php

```diff
@@ -26,6 +26,79 @@ function tevkori_get_picturefill() {
 add_action( 'wp_enqueue_scripts', 'tevkori_get_picturefill' );
 
 /**
+ * Return a source size list for an image from an array of values
+ *
+ * @since 2.2.0
+ *
+ * @param int $id 			Image attacment ID.
+ * @param string $size	Optional. Name of image size. Default value: 'thumbnail'.
+ * @param array $args {
+ *     Optional. Arguments to retrieve posts.
+ *
+ *     @type array|string 	$sizes     An array or string containing of size information.
+ * }
+ * @return string|bool A valid source size list as a `sizes` attribute or false.
+ */
+function tevkori_get_sizes_string( $id, $size, $args ) {
+	// See which image is being returned and bail if none is found
+	if ( ! $image = wp_get_attachment_image_src( $id, $size ) ) {
+		return false;
+	};
+
+	// Get the image width
+	$img_width = $image[1] . 'px';
+
+	// Set up our default values
+	$defaults = array(
+		'sizes' => array(
+			array(
+				'size_value' 	=> '100vw',
+				'mq_value'		=> $img_width,
+				'mq_name'			=> 'max-width'
+			),
+			array(
+				'size_value'	=> $img_width
+			),
+		)
+	);
+
+	$args = wp_parse_args( $args, $defaults );
+
+	// If sizes is passed as a string, just use the string
+	if ( is_string( $args['sizes'] ) ) {
+		$size_list = $args['sizes'];
+
+	// Otherwise, breakdown the array and build a sizes string
+	} elseif ( is_array( $args['sizes'] ) ) {
+
+		foreach( $args['sizes'] as $size ) {
+			// Use 100vm as the size value unless something else is specified.
+			$size_value = ( $size['size_value'] ) ? $size['size_value'] : '100vm';
```

Comment:
this should probably be vw no? vm units don't seem to have great support http://caniuse.com/#search=vm